pyartifactory/objects.py

```diff
@@ -419,25 +419,25 @@ def get_repo(
         LocalRepositoryResponse, VirtualRepositoryResponse, RemoteRepositoryResponse
     ]:
         """
-        Finds repository in artifactory. Fill object if exist
+        Finds repository in artifactory. Raises an exception if the repo doesn't exist.
         :param repo_name: Name of the repository to retrieve
-        :return: LocalRepositoryResponse, VirtualRepositoryResponse or RemoteRepositoryResponse object
+        :return: Either a local, virtual or remote repository
         """
         try:
             response = self._get(f"api/{self._uri}/{repo_name}")
-            repo = parse_obj_as(BaseRepositoryModel, response.json())
-            found_repo: Union[
+            repo: Union[
```

Comment:
You don't need to be explicit about the typing here, `parse_obj_as` already returns the type passed in the first parameter (see [here](https://github.com/samuelcolvin/pydantic/blob/9982eb43ddfdb2f112976be390b6912973bd95d0/pydantic/tools.py#L33))

Reply:
I got an error from mypy if I don't explicit the type. Mypy thinks that repo is of type any